File: apps/customer-mobile/lib/data/api/menu_api.dart
import '../dto/menu_dto.dart';
import 'api_client.dart';

class MenuApi {
  const MenuApi(this._client);

  final ApiClient _client;

  Future<List<CategoryDto>> listCategories(String tenantId) async {
    final json = await _client.getList('/api/v1/menu/categories', tenantId: tenantId);
    return json.map((e) => CategoryDto.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<List<ProductDto>> listProducts(String tenantId) async {
    final json = await _client.getList('/api/v1/menu/products', tenantId: tenantId);
    return json.map((e) => ProductDto.fromJson(e as Map<String, dynamic>)).toList();
  }
}

File: apps/customer-web/src/infrastructure/api/menuApi.ts
import { apiRequest } from './httpClient';

const MENU_SERVICE_URL = import.meta.env.VITE_MENU_SERVICE_URL;

export interface CategoryDto {
  id: string;
  tenant_id: string;
  name: string;
  display_order: number;
  is_active: boolean;
}

export interface ProductDto {
  id: string;
  tenant_id: string;
  category_id: string;
  name: string;
  description: string;
  price: number;
  cost_price: number;
  tax_rate: number;
  photo_url: string;
  display_order: number;
  is_active: boolean;
}

export async function listCategories(tenantId: string): Promise<CategoryDto[]> {
  return apiRequest<CategoryDto[]>(MENU_SERVICE_URL, '/api/v1/menu/categories', { tenantId });
}

export async function listProducts(tenantId: string): Promise<ProductDto[]> {
  return apiRequest<ProductDto[]>(MENU_SERVICE_URL, '/api/v1/menu/products', { tenantId });
}
